Allow to customize AGP and Kotlin plugin versions

lptr opened the pull request.

lptr committed:
Allow to customize AGP and Kotlin plugin versions

wolfs commented on run-faster-incremental-java.sh:
```diff
@@ -1,9 +1,14 @@
 gradle-profiler/gradlew --project-dir gradle-profiler install
 
 gradle-profiler/build/install/gradle-profiler/bin/gradle-profiler \
+    -DagpVersion=$AGP_VERSION \
```
Why are you passing this system property to Gradle profiler? Don't you need to pass it to the santa tracker build?

lptr replied:
It passes them on as expected, apparently. How would I only pass it to the builds themselves?

wolfs replied:
I would have added something to the scenario file. I am actually surprised Gradle profiler passes system properties from the command line to the Gradle builds it runs. Is that documented somewhere?

wolfs replied:
Seems like the tooling API transfers all system properties to the daemon: https://discuss.gradle.org/t/system-property-leakage-via-the-tooling-api/19937 ?

lptr replied:
I don't know how to do this in a nice way in HOCON. It works now with the bash script, so I'd leave it as is.

I'd argue that the scenario file should handle things _inside_ Gradle, while the Bash scripts should handle the environment.

wolfs replied:
You can access environment variables in HOCON: https://github.com/lightbend/config/blob/master/HOCON.md#substitution-fallback-to-environment-variables

It seems to me that the `agpVersion` should be part of the scenario file, then we would at least see in the gradle profiler log which AGP version is used, since it would be a parameter to the Gradle invocation.

Your argument is that the bash script should track the Gradle and the AGP versions? I can live with that as well.

lptr replied:
That works the other way around: you can use the environment as a fallback. What I'd need is to take the environment variable's value, and fall back to some other value if not specified.

Anyway, let's leave it like this for now, and change it if it causes problems.

wolfs commented on run-faster-incremental-kotlin.sh:
```diff
@@ -1,9 +1,16 @@
 
 gradle-profiler/build/install/gradle-profiler/bin/gradle-profiler \
+    -DagpVersion=$AGP_VERSION \
+    -DkotlinVersion=$KOTLIN_VERSION \
```
Same question as above.

lptr committed:
Enable all benchmarks for Kotlin version, too

lptr committed:
Upgrade Gradle Profiler to 0.11

lptr committed:
Do not measure partial invalidation

The effect is usually small, and we don't really care about it separately anyway.

lptr committed:
Don't run IE tests with Kotlin for now

There seems to be some issue causing intermittent problems.

lptr committed:
Add order numbers to titles

This allows the results to be sorted more meaningfully.

wolfs commented on santa-tracker-base.conf:
```diff
@@ -37,31 +37,31 @@ enableVfsRetention = [
-onlyInstantExecution = ${noOptimizations} {
-    title = "Instant execution"
-    gradle-args = ${noOptimizations.gradle-args} ${enableInstantExecution}
+onlyPartialInvalidation = ${noOptimizations} {
```
I wonder if we should get rid of this scenario as well.

lptr replied:
I removed it.

lptr committed:
Remove 'onlyPartialInvalidation' scenario
